lib/provider.js

```diff
@@ -115,15 +115,45 @@ const mergeConfigs = (providerConfig) => {
  * @param file
  * @returns string
  */
-const generateUploadFileName = (basePath, file) => {
-  const backupPath =
-    file.related && file.related.length > 0 && file.related[0].ref
-      ? `${file.related[0].ref}`
-      : `${file.hash}`;
-  const filePath = file.path ? `${file.path}/` : `${backupPath}/`;
-  const extension = file.ext.toLowerCase();
-  const fileName = slugify(path.basename(file.name + '_' + file.hash, file.ext)) + extension;
-  return `${basePath}${filePath}${fileName}`;
+const generateUploadFileName = (basePath, file, shortFilePaths) => {
+  if (!shortFilePaths) {
+    // Default version
+    const backupPath =
+      file.related && file.related.length > 0 && file.related[0].ref
+        ? `${file.related[0].ref}`
+        : `${file.hash}`;
+    const filePath = file.path ? `${file.path}/` : `${backupPath}/`;
+
+    // make sure that the extension is correct and the filename is valid:
+    // path.basename -> https://nodejs.org/docs/latest/api/path.html#path_path_basename_path_ext
+    // slugify -> https://www.npmjs.com/package/slugify
+    const fileName =
+      slugify(path.basename(file.name + '_' + file.hash, file.ext)) + file.ext.toLowerCase();
+
+    return `${basePath}${filePath}${fileName}`;
```

Comment:
If it returns you don't need to `else` it :wink:

Reply:
Yes that is true. Looked better because the regular version came first. Even better will be if I wrap the new version in a single `if` block and the regular version afterwards.